statecheck/extract_bool_value_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package statecheck_test
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	r "github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/statecheck"
+	"github.com/hashicorp/terraform-plugin-testing/tfjsonpath"
+)
+
+func TestExtractBoolValue_CheckState_Basic(t *testing.T) {
+	t.Parallel()
+
+	// targetVar will be set to the extracted value.
+	var targetVar bool
+
+	r.Test(t, r.TestCase{
+		ProviderFactories: map[string]func() (*schema.Provider, error){
+			"test": func() (*schema.Provider, error) { //nolint:unparam // required signature
+				return testProvider(), nil
+			},
+		},
+		Steps: []r.TestStep{
+			{
+				Config: `resource "test_resource" "one" {
+					bool_attribute = false
+				}
+				`,
+				ConfigStateChecks: []statecheck.StateCheck{
+					statecheck.ExtractBoolValue(
+						"test_resource.one",
+						tfjsonpath.New("bool_attribute"),
+						&targetVar,
+					),
+				},
+				Check: testAccAssertBoolEquals(false, targetVar),
```

Comment:
Unfortunately, it appears that `Checks` are ran before `ConfigStateChecks`, which makes this not possible.